lib/iris/cube.py

```diff
@@ -2013,24 +2015,32 @@ def subset(self, coord):
 
         # Get the coord to extract from the cube
         coord_to_extract = self.coord(coord)
-        if len(self.coord_dims(coord_to_extract)) > 1:
-            msg = "Currently, only 1D coords can be used to subset a cube"
-            raise iris.exceptions.CoordinateMultiDimError(msg)
-        # Identify the dimension of the cube which this coordinate references
-        coord_to_extract_dim = self.coord_dims(coord_to_extract)[0]
-
-        # Identify the indices which intersect the requested coord and
-        # coord_to_extract
-        coordinate_indices = coord_to_extract.intersect(coord,
-                                                        return_indices=True)
-
-        # Build up a slice which spans the whole of the cube
-        full_slice = [slice(None, None)] * len(self.shape)
-        # Update the full slice to only extract specific indices which were
-        # identified above
-        full_slice[coord_to_extract_dim] = coordinate_indices
-        full_slice = tuple(full_slice)
-        return self[full_slice]
+
+        # If scalar, return the whole cube. Not possible to subset 1 point.
+        if coord_to_extract in self.aux_coords and\
+                len(coord_to_extract.points) == 1:
+            result = self
```

Comment:
This could do with some logic to check the actual *value* of the supplied coordinate and mimic the found/not-found behaviour the comes with non-scalar coordinates.